src/components/controller.cpp

```diff
@@ -460,10 +460,30 @@ bool controller::process_update(bool force) {
   const bar_settings& bar{m_bar->settings()};
   string contents;
   string separator{bar.separator};
-  string padding_left(bar.padding.left, ' ');
-  string padding_right(bar.padding.right, ' ');
-  string margin_left(bar.module_margin.left, ' ');
-  string margin_right(bar.module_margin.right, ' ');
+
+  auto add_surrounding_tag = [](const space_size& space_size) -> std::string {
```

Comment:
This functionality is now implemented twice, once here and once in the builder. It probably makes sense to create a single function that generates either a number of spaces or an apropriate `%{O` tag.

I'm not sure about the apropriate place for this function. I generally would like to the builder be responsible, so a static function there may make sense. But this would couple the builder and the controller together but that may not be too bad since the controller also *builds* the module output.

Reply:
Fixed by https://github.com/jaagr/polybar/pull/1671/commits/f2c170c6c69522461f808ba0a762430366dc192e